src/java/grl/src/grl/driver/GRL_Driver.java

```diff
@@ -379,84 +412,48 @@ else if (_currentKUKAiiwaState.armControlState().stateType() == grl.flatbuffer.A
 				if (currentMotion != null && _commandInterface!=grl.flatbuffer.KUKAiiwaInterface.FRI) currentMotion.cancel();
 				if(!cancelTeachMode()) continue;
 
+		        // TODO: switch control mode in config change check? or check in individual armControlState?
```

Comment:
The issue with switching in config is that currentMotion will be initialized. Unless you want to pass NO_CONTROL as default to EControlMode. And  switching to MoveArmJointServo sets it to appropriate EControlMode.